Polyfills/Console/Source/Console.cpp

```diff
@@ -34,16 +34,82 @@ namespace
         }
     }
 
+    inline bool isNumberEnding(char c)
+    {
+        return c == 'd' || c == 'i' || c == 'f';
+    }
+
+    inline bool isStringOrObjectEnding(char c)
+    {
+        return c == 'o' || c == 'O' || c == 's';
+    }
+
     void InvokeCallback(Babylon::Polyfills::Console::CallbackT callback, const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel)
     {
-        std::stringstream ss{};
-        for (size_t i = 0; i < info.Length(); i++)
+        std::ostringstream ss{};
+        if (info.Length() > 0)
         {
-            if (i > 0)
+            std::string firstArg = info[0].ToString();
+            size_t currArgIndex = 1;
+
+            std::size_t j = 0;
+            while (j < firstArg.size())
+            {
+                const char currChar = firstArg.at(j);
+                // When a '%' is encountered, check the next character to determine the type of string we have
+                if (currChar == '%' && j < firstArg.size() - 1 && currArgIndex < info.Length())
+                {
+                    char nextChar = firstArg.at(j + 1);
+                    Napi::Value currArg = info[currArgIndex];
+                    // the next character can be one of: [soO], when the substitution string specifies a string
+                    if (isStringOrObjectEnding(nextChar))
+                    {
+                        ss << currArg.ToString().Utf8Value();
+                        currArgIndex++;
+                    }
+                    // or [dif], when it specifies a number
+                    else if (isNumberEnding(nextChar))
+                    {
+                        double d = currArg.ToNumber().DoubleValue();
```

Comment:
Do we need to check if `currArg` is actually a number before forcing it to a number?

Reply:
If currArg is not a number, it's converted to NaN which is what we want